### DISCUSSION TITLE
Android: why `notificationReceived` event is not triggered in case of background app and notification from firebase notification composer

### DISCUSSION THREAD

**Discussion by StefanZivkovic** [capawesome-team/capacitor-firebase#498] [Q&A] [answered]
android, app in background. Notification is sent from firebase console composer, so it looks like this, when inspecting firebase campaigns POST endpoint response
```
dataBundle: {_body: "_ddd", _title: "_tt"}
displayName: "zxcz",
name: "projects/.../campaigns/..."
notificationOptions: {messageText: "zxc", messageTitle: "zxcz", expiryTime: "2419200s"}
startTime: "2023-11-20T12:15:53.065036Z"
state: "RUNNING"
targetTime: {year: 2023, month: 11, day: 20, hours: 4, minutes: 15, seconds: 53,…}
targetingCondition: "app.id == '1:...:android:...'"
targetingConditionJson: "{\"or\":{\"conditions\":[{\"and\":{\"conditions\":[{\"appId\":{\"targetGmpAppId\":\"1:...:android:...\"}}]}}]}}"
```

Why is this the issue - is this considered as a `notification` message - hence event `notificationReceived` is not trigger for that reason (when app is in background)?

https://firebase.google.com/docs/cloud-messaging/js/send-with-console
![image](https://github.com/capawesome-team/capacitor-firebase/assets/44195291/7866d9b4-939b-43ce-8b4e-b9641f4c332a)

I saw in docs of this plugin that it mentions that event in background app will only be triggered in case when some custom data is available (for android and for ios), but anyway i feel a bit in a blurry here - would like to understand more. I guess its not question for you directly, but maybe you can share some references with some good explanations. 

Is this limitation of hybrid apps, would it be possible with pure native code/apps?

**Comment by robingenz** [comment, marked as answer]
Thank you for your request!
Your code is truncated and therefore does not contain much information. But yes, as far as I know, you can't send "background notifications" via the Firebase Console.

> Is this limitation of hybrid apps, would it be possible with pure native code/apps?

This is a general limitation of Firebase. You can find out more here: https://firebase.google.com/docs/cloud-messaging/android/receive#handling_messages

**Reply by StefanZivkovic** [reply]
It is not my code, it is pasted response content from browser->network->response when do `publish` button in `campaigns` in firebase notification composer interface. My idea was to point out `keys` in response, mostly on `dataBundle` and `notificationOptions`, but it might not have to be related with anything, i was just trying to provide more input and hence maybe hoping for more feedback :) 

While i am well aware that firebase admin sdk implementation on my backend service might handle this, I would also need to add that same UI for campaings feature which already exists in firebase console. But thats another thing. On the other hand, i have gone through that docs link many times and in docs in general and I did not have clear impression that this use case is a limitation (maybe because there is no clear explanation in firebase docs for firebase composer - whether sending campaigns is notification or data message. Or maybe its there, but i misread it....). 

Also i was aware of these differences written for data and notification messages., but somehow I guess I have always hoped it could be tweaked by some java/swift tricks, or maybe handled with some background (capacitor plugin) runners...

Again, thanks for response. :pray:
